contracts/Witnet.sol

```diff
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: MIT
 
-pragma solidity 0.6.12;
+pragma solidity 0.8.4;
 pragma experimental ABIEncoderV2;
```

Comment:
As in `contracts/CBOR.sol': no need to include this pragma on ^0.8.0, as ABIEncoderV2 is activated by default. If you want to be specific, pragma abicoder v2; should be used nistead.

Reply:
According to Slither, the following methods should be declared as external (gas optimization):

- `isOk(Witnet.Result)`
- `isError(Witnet.Result)`
- `asBytes(Witnet.Result)`
- `asErrorCode(Witnet.Result)`
- `asErrorMessage(Witnet.Result)`
- `asBool(Witnet.Result)`
- `asFixed16(Witnet.Result)`
- `asFixed16Array(Witnet.Result)`
- `asInt128(Witnet.Result)`
- `asInt128Array(Witnet.Result)`
- `asString(Witnet.Result)`
- `asStringArray(Witnet.Result)`
- `asUint64(Witnet.Result)`
- `asUint64Array(Witnet.Result)`